commit 59e205a57a04fced6bb7b8ec0b5dec29461a9996
Author: Blake Embrey <hello@blakeembrey.com>
Date:   Mon Jun 15 19:45:22 2026 -0700

    Upgrade `content-disposition` (#7233)
    
    * Upgrade content-disposition
    
    * Better backward compat
    
    * chore: upgrade content-disposition to 2.0.1 and update History.md
    
    * test: add case for encoding latin1 file names with ASCII fallback in attachment
    
    * test: add cases for quoting invalid file names and handling empty file names in attachment
    
    ---------
    
    Co-authored-by: Sebastian Beltran <bjohansebas@gmail.com>

diff --git a/test/acceptance/downloads.js b/test/acceptance/downloads.js
index 6db43b35..3deed3e0 100644
--- a/test/acceptance/downloads.js
+++ b/test/acceptance/downloads.js
@@ -15,7 +15,7 @@ describe('downloads', function(){
     it('should have a download header', function (done) {
       request(app)
         .get('/files/notes/groceries.txt')
-        .expect('Content-Disposition', 'attachment; filename="groceries.txt"')
+        .expect('Content-Disposition', 'attachment; filename=groceries.txt')
         .expect(200, done)
     })
   })
@@ -24,7 +24,7 @@ describe('downloads', function(){
     it('should have a download header', function(done){
       request(app)
       .get('/files/amazing.txt')
-      .expect('Content-Disposition', 'attachment; filename="amazing.txt"')
+      .expect('Content-Disposition', 'attachment; filename=amazing.txt')
       .expect(200, done)
     })
   })
diff --git a/test/res.attachment.js b/test/res.attachment.js
index 8644bab5..84581bd6 100644
--- a/test/res.attachment.js
+++ b/test/res.attachment.js
@@ -31,7 +31,33 @@ describe('res', function(){
 
       request(app)
       .get('/')
-      .expect('Content-Disposition', 'attachment; filename="image.png"', done);
+      .expect('Content-Disposition', 'attachment; filename=image.png', done);
+    })
+
+    it('should quote file names that are not a valid token', function(done){
+      var app = express();
+
+      app.use(function(req, res){
+        res.attachment('/path/to/my report.png');
+        res.send('foo');
+      });
+
+      request(app)
+      .get('/')
+      .expect('Content-Disposition', 'attachment; filename="my report.png"', done);
+    })
+
+    it('should handle an empty file name', function(done){
+      var app = express();
+
+      app.use(function(req, res){
+        res.attachment('');
+        res.send('foo');
+      });
+
+      request(app)
+      .get('/')
+      .expect('Content-Disposition', 'attachment; filename=""', done);
     })
 
     it('should set the Content-Type', function(done){
@@ -63,6 +89,20 @@ describe('res', function(){
       .expect(200, done);
     })
 
+    it('should encode latin1 file names with an ASCII fallback and filename* param', function(done){
+      var app = express();
+
+      app.use(function(req, res){
+        res.attachment('/locales/café.txt');
+        res.send('coffee');
+      });
+
+      request(app)
+      .get('/')
+      .expect('Content-Disposition', 'attachment; filename="caf?.txt"; filename*=UTF-8\'\'caf%C3%A9.txt')
+      .expect(200, done);
+    })
+
     it('should set the Content-Type', function(done){
       var app = express();
 
diff --git a/test/res.download.js b/test/res.download.js
index e9966007..f059ad13 100644
--- a/test/res.download.js
+++ b/test/res.download.js
@@ -24,7 +24,7 @@ describe('res', function(){
       request(app)
       .get('/')
       .expect('Content-Type', 'text/html; charset=utf-8')
-      .expect('Content-Disposition', 'attachment; filename="user.html"')
+      .expect('Content-Disposition', 'attachment; filename=user.html')
       .expect(200, '<p>{{user.name}}</p>', done)
     })
 
@@ -67,7 +67,7 @@ describe('res', function(){
       request(app)
       .get('/')
       .expect('Content-Type', 'text/html; charset=utf-8')
-      .expect('Content-Disposition', 'attachment; filename="document"')
+      .expect('Content-Disposition', 'attachment; filename=document')
       .expect(200, done)
     })
   })
@@ -84,7 +84,7 @@ describe('res', function(){
       request(app)
       .get('/')
       .expect('Content-Type', 'text/html; charset=utf-8')
-      .expect('Content-Disposition', 'attachment; filename="user.html"')
+      .expect('Content-Disposition', 'attachment; filename=user.html')
       .expect(200, cb);
     })
 
@@ -113,7 +113,7 @@ describe('res', function(){
         request(app)
           .get('/')
           .expect('Content-Type', 'text/plain; charset=utf-8')
-          .expect('Content-Disposition', 'attachment; filename="name.txt"')
+          .expect('Content-Disposition', 'attachment; filename=name.txt')
           .expect(200, 'tobi', cb)
       })
 
@@ -162,7 +162,7 @@ describe('res', function(){
       request(app)
         .get('/')
         .expect(200)
-        .expect('Content-Disposition', 'attachment; filename=".name"')
+        .expect('Content-Disposition', 'attachment; filename=.name')
         .expect('Cache-Control', 'public, max-age=14400')
         .expect(utils.shouldHaveBody(Buffer.from('tobi')))
         .end(done)
@@ -259,7 +259,7 @@ describe('res', function(){
           request(app)
             .get('/')
             .expect(200)
-            .expect('Content-Disposition', 'attachment; filename="user.html"')
+            .expect('Content-Disposition', 'attachment; filename=user.html')
             .end(done)
         })
 
@@ -277,7 +277,7 @@ describe('res', function(){
           request(app)
             .get('/')
             .expect(200)
-            .expect('Content-Disposition', 'attachment; filename="user.html"')
+            .expect('Content-Disposition', 'attachment; filename=user.html')
             .end(done)
         })
       })
@@ -296,7 +296,7 @@ describe('res', function(){
         request(app)
           .get('/')
           .expect(200)
-          .expect('Content-Disposition', 'attachment; filename="name.txt"')
+          .expect('Content-Disposition', 'attachment; filename=name.txt')
           .expect(utils.shouldHaveBody(Buffer.from('tobi')))
           .end(done)
       })
@@ -313,7 +313,7 @@ describe('res', function(){
         request(app)
           .get('/')
           .expect(200)
-          .expect('Content-Disposition', 'attachment; filename="name.txt"')
+          .expect('Content-Disposition', 'attachment; filename=name.txt')
           .expect(utils.shouldHaveBody(Buffer.from('tobi')))
           .end(done)
       })
@@ -367,7 +367,7 @@ describe('res', function(){
       request(app)
       .get('/')
       .expect('Content-Type', 'text/html; charset=utf-8')
-      .expect('Content-Disposition', 'attachment; filename="document"')
+      .expect('Content-Disposition', 'attachment; filename=document')
       .expect(200, cb);
     })
   })
@@ -386,7 +386,7 @@ describe('res', function(){
       .get('/')
       .expect(200)
       .expect('Content-Type', 'text/html; charset=utf-8')
-      .expect('Content-Disposition', 'attachment; filename="document"')
+      .expect('Content-Disposition', 'attachment; filename=document')
       .end(cb)
     })
 
@@ -403,7 +403,7 @@ describe('res', function(){
       request(app)
         .get('/')
         .expect(200)
-        .expect('Content-Disposition', 'attachment; filename="document"')
+        .expect('Content-Disposition', 'attachment; filename=document')
         .expect('Cache-Control', 'public, max-age=14400')
         .expect(utils.shouldHaveBody(Buffer.from('tobi')))
         .end(done)
@@ -426,7 +426,7 @@ describe('res', function(){
         .get('/')
         .expect(200)
         .expect('Content-Type', 'text/x-custom')
-        .expect('Content-Disposition', 'attachment; filename="document"')
+        .expect('Content-Disposition', 'attachment; filename=document')
         .end(done)
       })
 
@@ -446,7 +446,7 @@ describe('res', function(){
         .get('/')
         .expect(200)
         .expect('Content-Type', 'text/x-custom')
-        .expect('Content-Disposition', 'attachment; filename="document"')
+        .expect('Content-Disposition', 'attachment; filename=document')
         .end(done)
       })
     })
